probe-rs/src/probe.rs

```diff
@@ -946,9 +946,9 @@ pub enum DebugProbeSelectorParseError {
 /// assert_eq!(selector.vendor_id, 0x1942);
 /// assert_eq!(selector.product_id, 0x1337);
 /// ```
-#[derive(Debug, Clone, Serialize, Deserialize)]
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash)]
 // We need this so that serde will first convert from the string `VID:PID:<Serial>` to a struct before deserializing.
-#[serde(try_from = "String")]
+#[serde(try_from = "String", into = "String")]
```

Comment:
We should stop doing this

Reply:
This is how we read make serialization from the CLI work. Optimally we change this, indeed.